build/v1/types.go

```diff
@@ -740,6 +740,14 @@ type DockerBuildStrategy struct {
 	// policy. An additional experimental policy 'SkipLayersAndWarn' is the same as
 	// 'SkipLayers' but simply warns if compatibility cannot be preserved.
 	ImageOptimizationPolicy *ImageOptimizationPolicy `json:"imageOptimizationPolicy,omitempty" protobuf:"bytes,8,opt,name=imageOptimizationPolicy,casttype=ImageOptimizationPolicy"`
+
+	// volumes is a list of input volumes that can be mounted into buildah's runtime environment.
+	// Only a subset of Kubernetes Volume sources are supported by builds.
+	// More info: https://kubernetes.io/docs/concepts/storage/volumes
+	Volumes []BuildVolume `json:"volumes,omitempty" protobuf:"bytes,9,opt,name=volumes"`
```

Comment:
Would it make sense to have an encapsulating struct for the two arrays here?  I think it makes sense but will acquiesce to you two @adambkaplan @coreydaley if you all disagree.

My reasoning:  This is a bit different in the k8s case where volumes are defined at the pod spec level but volume mounts are at the container level.

For use, we have stuff at the same level, and certainly we don't envisioning using 1 array without the other.

Reply:
Makes sense - so our API could look like this:

```yaml
volumes:
- name: npm-secret
  secret:
    name: npm-secret
  mount:
    destinationPath: /var/run/secrets/npm-secret
    readOnly: true
```

@bparees thoughts?

Reply:
it does seem to remove a potential source of error (mismatched volume/volumemount names).

the only downside i can think of offhand is that you can't do something like mount a single volume in two different locations if you wanted to (without explicitly defining the volume twice also).  But i can't see that being critical.

so i think i'm +1 on @gabemontero's suggestion.

Reply:
How about:

```
volumes:
- name: npm-secret
  source:
	type: [secret/configmap]
  	secret:
    	    name: npm-secret
  mount:
    destinationPath: /var/run/secrets/npm-secret
    readOnly: true
```

Reply:
> the only downside i can think of offhand is that you can't do something like mount a single volume in two different locations if you wanted to (without explicitly defining the volume twice also). But i can't see that being critical.

Could this become an issue down the road if someone wants to mount a PVC in two locations? I think yes.

So to take @coreydaley 's suggestion (which I like), we can change `mount` to a `mounts` array, keyed off of the destination path:

```yaml
volumes:
- name: npm-secret
  source:
    type: [secret/configmap]
    secret:
      name: npm-secret
  mounts:
  -  destinationPath: /var/run/secrets/npm-secret
     readOnly: true
```